app/src/main/java/fr/iut_rodez/pathpilot_android_client/signup/SignUpService.java

```diff
@@ -0,0 +1,105 @@
+package fr.iut_rodez.pathpilot_android_client.signup;
+
+import static fr.iut_rodez.pathpilot_android_client.util.Network.getRequestQueue;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.JsonObjectRequest;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import fr.iut_rodez.pathpilot_android_client.BuildConfig;
+import fr.iut_rodez.pathpilot_android_client.login.Login;
+import fr.iut_rodez.pathpilot_android_client.util.Network;
+import fr.iut_rodez.pathpilot_android_client.util.Popup;
+import fr.iut_rodez.pathpilot_android_client.util.VolleyErrorHandler;
+
+public class SignUpService {
+
+    private static final String LOGIN_URL = BuildConfig.API_BASE_URL + "auth/signup";
+
+    private static final String TAG = SignUpService.class.getSimpleName();
+    public static final String CLE_MAIL = "mail";
+
+    public static void signUp(SignUpInput signUpInput, Context context) {
+        Log.d(TAG, "API URL: " + LOGIN_URL);
+
+        Popup popup = new Popup(context);
+
+        ProgressDialog progressDialog = new ProgressDialog(context);
+        progressDialog.show();
+
+        RequestQueue requestQueue = getRequestQueue(context);
+        JSONObject signUpInputJson = signUpInput.toJson();
+
+        Log.d(TAG, "signUp: " + signUpInputJson);
+
+        JsonObjectRequest request = new JsonObjectRequest(Request.Method.POST, LOGIN_URL, signUpInputJson,
+                response -> {
+                    Log.d(TAG, "onResponse: " + response);
+                    progressDialog.dismiss();
+
+                    // Button to go to login page
+                    Popup.Button btnLogin = new Popup.Button("Login", (dialog, which) -> {
+                        dialog.dismiss();
+
+                        Intent intent = new Intent(context, Login.class);
+                        intent.putExtra(CLE_MAIL, signUpInput.mail);
+
+                        context.startActivity(intent);
+                    });
+                    // Button to dismiss the dialog
+                    Popup.Button btnOk = new Popup.Button("OK", (dialog, which) -> dialog.dismiss());
+
+                    // Show success dialog with the two buttons
+                    popup.showAlertDialog("Success", "Account created successfully", btnLogin, btnOk, null);
+                },
+                error -> {
+                    Log.e(TAG, "Error while sending request", error);
+                    progressDialog.dismiss();
+                    VolleyErrorHandler.handleError(context, error);
+                }
+        );
+
+        requestQueue.add(request);
+    }
+
+    public static class SignUpInput {
```

Comment:
Add signature there also

Reply:
Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/43/commits/13651282009dcc58f629be9db6b86f229a8c799b